### PR TITLE
A11y updates

### PR DESCRIPTION
**Pull request by dpvc**
This PR adds support for semantic-enrichment, complexity computations, collapsing complex math, and landing of SRE.  It also includes an updated lab (`v3-lab.html`) that is in a different layout, and allows selecting the various components.

It was difficult to get a branch to merge into that only showed my changes (because I needed changes from `core-update`, so it actually includes `Explorer.js`, `Region.js`, and your semantic lab (now obsolete) as new files.  There are a couple of changes to handle the new loading of SRE, but nothing else.  I've included `sre.js`, which loads SRE (if it isn't already loaded) and provides a promise that is resolves when SRE is loaded and ready, and rejects if it fails to load, so the is now used to coordinate with Explorer and semantic-enrichment.

The semantic-enrich and complexity extensions are designed to make them work as part of the components approach that is in a previous pull request (and I've tested that it will work with that), but the branches would have to be merged together to make that work.

I have not yet made the pull request for the `core-updates` branch, which is needed to make these new extensions possible (it allows easy extension of the Handler, MathDocument, and MathItem classes in order to add the methods needed for enrichment and complexity computations).  I will make a separate PR for that shortly.

### CODE REVIEW

**Comment by uptownnickbrown** [comment]
Hi @dpvc - we spoke last week about us possibly using mathjax-v3 to solve some of our challenges with server processing using v2 for our Bookshelf ebook reader. I'm starting to test the beta build and this a11y-components branch you suggested looking at this week.

Testing this branch I'm unable to get SRE or the explorer working (I'm doing my browser testing in your `v3-lab.html` file). It seems like `sre` is being referenced in `Explorer.ts` before it is successfully fetched with the loader in `sre.ts`. Here is where I'm getting the first error:

https://github.com/mathjax/mathjax-v3/pull/181/files#diff-db7139ebbcd32343467d31a20c19cc54R227

I added some logging and `sre` is undefined in that context, I think because the `checkSRE` test fails on the first pass and succeeds on the second for me. Maybe it's not waiting for the promise as expected?

I'll keep debugging but thought I'd drop a note in case you had an idea for a quick fix - I'm still learning my way around the project.

Thanks!

**Comment by dpvc** [comment]
@uptownnickbrown, I had fixed that earlier, but apparently forgot to push the commit.  I've done so, so try it again now.  The Explorer still doesn't work for me, but that is something I've asked Volker to look into.  But at least it doesn't crash now when you load the page.

**Comment by uptownnickbrown** [comment]
Yep, thanks for the quick reply - after pulling the change in 01cf445 I'm seeing the same thing as you. The `sre` errors are gone but the Explorer isn't working (and I'm not seeing speech actually get generated). Maybe a red herring but what I'm seeing is a flurry of these errors on MouseOver and MouseOut events when you try to interact with the equation:

```
Uncaught TypeError: Cannot read property 'unhighlight' of null
    at TagExplorer.HoverExplorer.MouseOver (Explorer.ts:418)
```

Happy to test more when there's an update.

**Comment by dpvc** [comment]
Yes, that's what I get as well.  Volker will have to figure the one out, I'm afraid.

@zorkow, can you look into this, when you get the chance?

File: mathjax3-ts/a11y/complexity/collapse.ts [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+            value: 12
+        }
+    };
+
+    /**
+     *  These are the characters to use for the various collapsed elements
+     *  (if an object, then semantic-role is used to get the character
+     *  from the object)
+     */
+    public marker: TypeRole<string> = {
```
[the remark this answers is not in the document]
Comment (dpvc) [reply mathjax3-ts/a11y/complexity/collapse.ts]: Note that `TypeRole` is defined as

```
export type TypeRole<T> = {[type: string]: T | {[role: string]: T}};
```

so this is correct.  It produces an object that has values that are either the give type, or a subject of keys with the given type.

[the remark this answers is not in the document]
Comment (zorkow) [reply mathjax3-ts/a11y/complexity/collapse.ts]: Got it. Somehow I thought `T` was only referring to the key.

File: mathjax3-ts/a11y/sre.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * The promise for loading the SRE library, if it is not already loaded
+ */
+const srePromise = (typeof sre === 'undefined' ? asyncLoad(SRELIB) : Promise.resolve());
+
+/**
+ * Values to control the polling for when SRE is ready
+ */
+const SRE_DELAY = 100;        // in milliseconds
+const SRE_TIMEOUT = 5 * 1000; // 5 seconds
```
[the remark this answers is not in the document]
Comment (dpvc) [reply mathjax3-ts/a11y/sre.ts]: I'm not sure why it should.  The code checks for whether SRE is ready and if not, it waits `SRE_DELAY` and tries again.  It gives up after `SRE_TIMEOUT` has passed.  If you are suggesting that `SRE_TIMEOUT` should be a multiple of `SRE_DELAY` or some such thing, then (1) it already *is* a multiple, and (2), the `setTimeout()` function is not very accurate in terms of time (and some browsers have recently made some changes that cause it to always delay at least one second in most cases).  So I'm not sure what the dependence is supposed to be.

So I guess my answer is "no".

**Pull request merged**